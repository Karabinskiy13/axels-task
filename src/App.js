/* eslint-disable no-unused-vars */
import React from 'react';

import { PictureList, ModalView } from './components';
import { GlobalStyle } from './styled/Global';

function App() {
  return (
    <div className="App">
      <ModalView />
      <PictureList />
    </div>
  );
}

export default App;
