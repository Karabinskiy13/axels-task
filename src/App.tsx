import React from 'react';
import { Route, Routes } from 'react-router-dom';
import { ReactRouter6Adapter } from 'use-query-params/adapters/react-router-6';
import { QueryParamProvider } from 'use-query-params';

import { PictureList, Chart } from './components';
import 'bootstrap/dist/css/bootstrap.min.css';
import Favorites from './components/Favorites';

const App = () => (
  <div className="App">
    <QueryParamProvider adapter={ReactRouter6Adapter}>
      <Routes>
        <Route path={'/'} element={<PictureList />} />
        <Route path={'/charts'} element={<Chart />} />
        <Route path={'/favorites'} element={<Favorites />} />
      </Routes>
    </QueryParamProvider>
  </div>
);

export default App;
