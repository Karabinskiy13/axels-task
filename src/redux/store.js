import { configureStore, combineReducers } from '@reduxjs/toolkit';
import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER
} from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import pictureReducer from '../redux/ducks/pictures';

const rootReducer = combineReducers({ pictureReducer });

const persistConfig = {
  key: 'root',
  storage
};
const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER]
      }
    })
});
export const persistor = persistStore(store);

export default store;
