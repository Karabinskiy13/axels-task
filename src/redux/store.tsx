import { configureStore, combineReducers } from '@reduxjs/toolkit';

import pictureReducer from './ducks/pictures';

const rootReducer = combineReducers({ pictureReducer });

const store = configureStore({
  reducer: rootReducer
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
export default store;
