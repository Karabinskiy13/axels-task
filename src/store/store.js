import { configureStore, combineReducers } from '@reduxjs/toolkit';

import pictureReducer from './slice/picture.slice';

const rootReducer = combineReducers({ pictureReducer });

const store = configureStore({
  reducer: rootReducer
});

export default store;
