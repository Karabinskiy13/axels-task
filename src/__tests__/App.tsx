import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { ReactRouter6Adapter } from 'use-query-params/adapters/react-router-6';
import { QueryParamProvider } from 'use-query-params';

import App from '../App';
import store from '../redux/store';

test('Should render App component', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <QueryParamProvider adapter={ReactRouter6Adapter}>
          <App />
        </QueryParamProvider>
      </BrowserRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});
