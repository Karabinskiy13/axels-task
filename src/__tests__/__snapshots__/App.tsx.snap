// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render App component 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div>
      <label
        class="sc-gsnTZi gcXMGB"
      >
        <label
          class="form__header form-label"
        >
          Picture Application
        </label>
      </label>
      <div
        class="sc-dkzDqf dklMiL"
      >
        <div
          class="ReactTags__tags react-tags-wrapper"
        >
          <p
            class="sr-only"
            role="alert"
            style="position: absolute; overflow: hidden; margin: -1px; padding: 0px; width: 1px; height: 1px; border: 0px;"
          />
          <div
            class="ReactTags__tagInput"
          >
            <input
              aria-label="Press enter to add new tag"
              class="ReactTags__tagInputField"
              data-automation="input"
              data-testid="input"
              placeholder="Press enter to add new tag"
              type="text"
              value=""
            />
          </div>
          <div
            class="ReactTags__selected"
          />
        </div>
      </div>
      <div
        class="container-fluid"
      >
        <div
          class="row"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
